paddle/fluid/memory/detail/memory_block.cc

```diff
@@ -27,117 +27,91 @@ void MemoryBlock::init(MetadataCache* cache, Type t, size_t index, size_t size,
                               static_cast<MemoryBlock*>(right_buddy)));
 }
 
-MemoryBlock::Type MemoryBlock::type(const MetadataCache& cache) const {
-  return cache.load(this).type;
+bool MemoryBlock::get_left_buddy(const MetadataCache& cache,
```

Comment:
Please follow google coding style. Do not parse non-const reference as output.

Reply:
Done